http-client-core/src/main/java/io/micronaut/http/client/BlockingHttpClient.java

```diff
@@ -140,12 +142,19 @@
         if (HttpStatus.class.isAssignableFrom(bodyType.getType())) {
             return (O) response.getStatus();
         } else {
-            return response
-                    .getBody()
-                    .orElseThrow(() -> new HttpClientResponseException(
-                            "Empty body",
-                            response
-                    ));
+            Optional<O> body = response.getBody();
+            Optional<byte[]> bodyBytes = response.getBody(Argument.of(byte[].class));
+            if (!body.isPresent() && bodyBytes.isPresent()) {
```

Comment:
Should be:

```java
if (!body.isPresent() && response.getBody(Argument.of(byte[].class)).isPresent()) {
```

To avoid the extra object creation and method call